pkg/gitparse/gitparse.go

```diff
@@ -147,8 +147,15 @@ func (d *Diff) write(ctx context.Context, p []byte) error {
 
 // finalize ensures proper closure of resources associated with the Diff.
 // handle the final flush in the finalize method, in case there's data remaining in the buffer.
+// Also set the hasDiffs flag on the commit to indicate that the commit has diffs.
+// This is needed to handle the case where a commit has no diffs, which would otherwise be missed.
 // This method should be called to release resources, especially when writing to a file.
-func (d *Diff) finalize() error { return d.contentWriter.CloseForWriting() }
+func (d *Diff) finalize() error {
+	if d.Commit != nil {
+		d.Commit.hasDiffs = true
+	}
```

Comment:
Why are you doing this here, instead of when the diff is created? (Maybe in `newDiff`, or even better, just at the `diff` call site.)